src/main/kotlin/parts/wisdom/arcraider/visualization/ARCWindow.kt

```diff
@@ -88,9 +94,35 @@ class ARCWindow(title: String, task: Task) : JFrame() {
         }
     }
 
-    private fun refresh(task: Task) {
+    private fun renderGame(task : Task, game : Game) {
+        val input = game.getInputGrid()
+        val output = game.getOutputGrid()
+        val curGrid = game.getCurrentGrid()
+
+        content.apply {
+            add(Box.createHorizontalBox().apply {
+                add(Box.createRigidArea(Dimension(GAP_SIZE_PX, 0)))
+                add(GridPane(input, false))
+                add(Box.createRigidArea(Dimension(GAP_SIZE_PX, 0)))
+                add(GridPane(curGrid, false))
+                add(Box.createHorizontalGlue())
+                add(Box.createRigidArea(Dimension(GAP_SIZE_PX * 2, 0)))
+                add(Box.createHorizontalGlue())
+                add(GridPane(output, true))
+                add(Box.createRigidArea(Dimension(GAP_SIZE_PX, 0)))
+            })
+        }
+
+    }
+
+    private fun refresh(task: Task, game: Game?) {
```

Comment:
I'd lean toward giving `game` a default value of `null`, since we call `refresh(...)` from points in the code that don't need to know about games, but your mileage may vary. Sometimes I do prefer not providing default arguments if I feel the caller should make a deliberate decision.

Reply:
added default argument